### ISSUE TITLE
pyhandle 1.1.1 fails to install on Centos 7.x using pip2

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2HANDLE/PYHANDLE#84]
When trying to install pyhandle 1.1.1 with pip2 it fails with the following error:
```
Yue are using pip version 8.1.2, however version 22.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting pyhandle
  Downloading https://files.pythonhosted.org/packages/c6/c1/f787154f5f9f2ae242c40875f12bc39b9bd98161f286252e77c80982ccbb/pyhandle-1.1.1.tar.gz (76kB)
    Complete output from command python setup.py egg_info:
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'long_description_content_type'
      warnings.warn(msg)
    error in pyhandle setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-kpCUCs/pyhandle/
You are using pip version 8.1.2, however version 22.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
The command '/bin/sh -c pip3 install -r /app/B2SAFE-core/scripts/tests/requirements.txt &&     pip2 install PyMySQL==0.10.1 &&     pip2 install pyhandle' returned a non-zero code: 1
```

**Comment by cookie33** [comment]
The previous version of pyhandle installed fine. This was version: 1.0.4
```
$ pip2 list | grep handle
...
pyhandle (1.0.4)
...
```

**Comment by cookie33** [comment]
Installing 1.0.4 using pip2
```
ollecting pyhandle==1.0.4
  Downloading https://files.pythonhosted.org/packages/a4/df/d500ba0a738e337b1c202919aec27050837e9f87eff81303c9f0e9a713a1/pyhandle-1.0.4.tar.gz (71kB)
Requirement already satisfied (use --upgrade to upgrade): requests in /usr/lib/python2.7/site-packages (from pyhandle==1.0.4)
Collecting datetime (from pyhandle==1.0.4)
  Downloading https://files.pythonhosted.org/packages/1d/bb/8e43c08cf8520816abe7afc5b5bff2a83807f371af90e7067d348b6a29ef/DateTime-4.5-py2.py3-none-any.whl (52kB)
Collecting future (from pyhandle==1.0.4)
  Downloading https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz (829kB)
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/site-packages (from pyhandle==1.0.4)
Requirement already satisfied (use --upgrade to upgrade): pymysql in /usr/lib/python2.7/site-packages (from pyhandle==1.0.4)
Collecting pytz (from datetime->pyhandle==1.0.4)
  Downloading https://files.pythonhosted.org/packages/60/2e/dec1cc18c51b8df33c7c4d0a321b084cf38e1733b98f9d15018880fb4970/pytz-2022.1-py2.py3-none-any.whl (503kB)
Collecting zope.interface (from datetime->pyhandle==1.0.4)
  Downloading https://files.pythonhosted.org/packages/47/55/20205df173d5c544a0002ee4b8f3aa862e3f1e2a0994a76ceed1587017f2/zope.interface-5.4.0-cp27-cp27mu-manylinux1_x86_64.whl (247kB)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python2.7/site-packages (from zope.interface->datetime->pyhandle==1.0.4)
Installing collected packages: pytz, zope.interface, datetime, future, pyhandle
  Running setup.py install for future: started
    Running setup.py install for future: finished with status 'done'
  Running setup.py install for pyhandle: started
    Running setup.py install for pyhandle: finished with status 'done'
Successfully installed datetime-4.5 future-0.18.2 pyhandle-1.0.4 pytz-2022.1 zope.interface-5.4.0
You are using pip version 8.1.2, however version 22.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

**Comment by merretbuurman** [comment]
Hi! I could not reproduce this. I used a CentOS 7 docker image, installed pip on it and tried to install pyhandle, which all worked perfectly. Have you tried updating pip?

My steps in trying to reproduce:

```
# Running a CentOS 7 test docker:
[me@localhost ~]$ docker run --name test -it centos:centos7.9.2009 /bin/bash
[root@9e6effe0556d /]# python --version
Python 2.7.5

# Installing pip:
[root@9e6effe0556d /]# curl -o get-pip.py https://bootstrap.pypa.io/pip/2.7/get-pip.py 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1863k  100 1863k    0     0   314k      0  0:00:05  0:00:05 --:--:--  424k
[root@9e6effe0556d /]# python get-pip.py 
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Collecting pip<21.0
  Downloading pip-20.3.4-py2.py3-none-any.whl (1.5 MB)
     |################################| 1.5 MB 6.2 MB/s 
Collecting setuptools<45
  Downloading setuptools-44.1.1-py2.py3-none-any.whl (583 kB)
     |################################| 583 kB 4.3 MB/s 
Collecting wheel
  Downloading wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-20.3.4 setuptools-44.1.1 wheel-0.37.1

# Installing pyhandle using pip:
[root@9e6effe0556d /]# pip install pyhandle
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Collecting pyhandle
  Downloading pyhandle-1.0.4.tar.gz (71 kB)
     |################################| 71 kB 119 kB/s 
Collecting requests
  Downloading requests-2.27.1-py2.py3-none-any.whl (63 kB)
     |################################| 63 kB 316 kB/s 
Collecting datetime
  Downloading DateTime-4.5-py2.py3-none-any.whl (52 kB)
     |################################| 52 kB 201 kB/s 
Collecting future
  Downloading future-0.18.2.tar.gz (829 kB)
     |################################| 829 kB 4.6 MB/s 
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting pymysql
  Downloading PyMySQL-0.10.1-py2.py3-none-any.whl (47 kB)
     |################################| 47 kB 1.3 MB/s 
Collecting certifi>=2017.4.17
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
     |################################| 149 kB 15.6 MB/s 
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.11-py2.py3-none-any.whl (139 kB)
     |################################| 139 kB 25.8 MB/s 
Collecting idna<3,>=2.5; python_version < "3"
  Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
     |################################| 58 kB 1.6 MB/s 
Collecting chardet<5,>=3.0.2; python_version < "3"
  Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
     |################################| 178 kB 21.6 MB/s 
Collecting pytz
  Downloading pytz-2022.1-py2.py3-none-any.whl (503 kB)
     |################################| 503 kB 5.5 MB/s 
Collecting zope.interface
  Downloading zope.interface-5.4.0-cp27-cp27mu-manylinux2010_x86_64.whl (247 kB)
     |################################| 247 kB 5.4 MB/s 
Requirement already satisfied: setuptools in /usr/lib/python2.7/site-packages (from zope.interface->datetime->pyhandle) (44.1.1)
Building wheels for collected packages: pyhandle, future
  Building wheel for pyhandle (setup.py) ... done
  Created wheel for pyhandle: filename=pyhandle-1.0.4-py2.py3-none-any.whl size=93608 sha256=5de55ee53495e3491b1d2e8fcd9e80b96e4d2464dc16135314e8d060b191cfd2
  Stored in directory: /root/.cache/pip/wheels/ff/30/34/9ef4e0391a8e7438f37dc5bba62f9257474347759c9b3a7211
  Building wheel for future (setup.py) ... done
  Created wheel for future: filename=future-0.18.2-py2-none-any.whl size=502566 sha256=83c0e3994adfefd64a03013891be36ad2ce2771afbfb980d969fd665b5ed79d0
  Stored in directory: /root/.cache/pip/wheels/5f/11/0c/aad680baf5ef4fbcbab992c9f03e1130357e0c173a4fdabfff
Successfully built pyhandle future
Installing collected packages: certifi, urllib3, idna, chardet, requests, pytz, zope.interface, datetime, future, six, pymysql, pyhandle
  Attempting uninstall: chardet
    Found existing installation: chardet 2.2.1
    Uninstalling chardet-2.2.1:
      Successfully uninstalled chardet-2.2.1
Successfully installed certifi-2021.10.8 chardet-4.0.0 datetime-4.5 future-0.18.2 idna-2.10 pyhandle-1.0.4 pymysql-0.10.1 pytz-2022.1 requests-2.27.1 six-1.16.0 urllib3-1.26.11 zope.interface-5.4.0

# Importing it works fine:
[root@9e6effe0556d /]# python
Python 2.7.5 (default, Oct 14 2020, 14:45:30) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyhandle
>>> 
```

**Comment by cookie33** [comment]
Hi,

We install pip in a more regular way in our docker image:
```
yum install python2-pip
```

And you test and get version 1.0.4. While we try it and get version 1.1.1.
And that fails to install.

It can be seen in the following raw data of our docker image installation:
```
Step 11/14 : RUN pip3 install -r /app/B2SAFE-core/scripts/tests/requirements.txt &&     pip2 install PyMySQL==0.10.1 &&     pip2 install pyhandle
 ---> Running in 6177feb9ee33
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting defusedxml==0.4.1 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/09/3b/b1afa9649f48517d027e99413fec54f387f648c90156b3cf6451c8cd45f9/defusedxml-0.4.1.tar.gz (48kB)
Collecting funcsigs==0.4 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/5e/9f/025d4c92c6a1a94313cdf0813cd76f5700f8e5434fa15165090a6446ae22/funcsigs-0.4-py2.py3-none-any.whl
Collecting httplib2==0.9.2 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/ff/a9/5751cdf17a70ea89f6dde23ceb1705bfb638fd8cee00f845308bf8d26397/httplib2-0.9.2.tar.gz (205kB)
Collecting lxml (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/0f/4e/e139648475c0eb42cdb813461ae11f567e1888a20d7dfd5256f47e32909d/lxml-4.9.1-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (5.6MB)
Collecting mock==1.0.1 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 5))
  Downloading https://files.pythonhosted.org/packages/a2/52/7edcd94f0afb721a2d559a5b9aae8af4f8f2c79bc63fdbe8a8a6c9b23bbe/mock-1.0.1.tar.gz (818kB)
Collecting pbr==1.8.1 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 6))
  Downloading https://files.pythonhosted.org/packages/fc/37/94af8387babb09796d306b18cf94ee5c70388c875a16d8a88e471500452c/pbr-1.8.1-py2.py3-none-any.whl (89kB)
Collecting queuelib==1.4.2 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 7))
  Downloading https://files.pythonhosted.org/packages/16/4f/b307fc978a21bfbb138e8e01a9f4953191d439e30578f5e4fd5befa77de1/queuelib-1.4.2-py2.py3-none-any.whl
Collecting simplejson==3.8.1 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 8))
  Downloading https://files.pythonhosted.org/packages/70/b0/c8169f6c2b7927614bea25636485992248ae0b1d7bc83858a9185461f57b/simplejson-3.8.1.tar.gz (76kB)
Collecting six==1.10.0 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 9))
  Downloading https://files.pythonhosted.org/packages/c8/0a/b6723e1bc4c516cb687841499455a8505b44607ab535be01091c0f24f079/six-1.10.0-py2.py3-none-any.whl
Collecting wheel==0.24.0 (from -r /app/B2SAFE-core/scripts/tests/requirements.txt (line 10))
  Downloading https://files.pythonhosted.org/packages/e8/14/eaaac12b0432c984a27ad0050a5a0bedc6135df35a0f5f1f35228faf12d4/wheel-0.24.0-py2.py3-none-any.whl (63kB)
Installing collected packages: defusedxml, funcsigs, httplib2, lxml, mock, pbr, queuelib, simplejson, six, wheel
  Found existing installation: defusedxml 0.7.1
    Uninstalling defusedxml-0.7.1:
      Successfully uninstalled defusedxml-0.7.1
  Running setup.py install for defusedxml: started
    Running setup.py install for defusedxml: finished with status 'done'
  Running setup.py install for httplib2: started
    Running setup.py install for httplib2: finished with status 'done'
  Running setup.py install for mock: started
    Running setup.py install for mock: finished with status 'done'
  Running setup.py install for simplejson: started
    Running setup.py install for simplejson: finished with status 'done'
  Found existing installation: six 1.16.0
    Uninstalling six-1.16.0:
      Successfully uninstalled six-1.16.0
Successfully installed defusedxml-0.4.1 funcsigs-0.4 httplib2-0.9.2 lxml-4.9.1 mock-1.0.1 pbr-1.8.1 queuelib-1.4.2 simplejson-3.8.1 six-1.10.0 wheel-0.24.0
Collecting PyMySQL==0.10.1
  Downloading https://files.pythonhosted.org/packages/1a/ea/dd9c81e2d85efd03cfbf808736dd055bd9ea1a78aea9968888b1055c3263/PyMySQL-0.10.1-py2.py3-none-any.whl (47kB)
Installing collected packages: PyMySQL
Successfully installed PyMySQL-0.10.1
You are using pip version 8.1.2, however version 22.2.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting pyhandle
  Downloading https://files.pythonhosted.org/packages/c6/c1/f787154f5f9f2ae242c40875f12bc39b9bd98161f286252e77c80982ccbb/pyhandle-1.1.1.tar.gz (76kB)
    Complete output from command python setup.py egg_info:
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'long_description_content_type'
      warnings.warn(msg)
    error in pyhandle setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-8kIetg/pyhandle/
You are using pip version 8.1.2, however version 22.2.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
The command '/bin/sh -c pip3 install -r /app/B2SAFE-core/scripts/tests/requirements.txt &&     pip2 install PyMySQL==0.10.1 &&     pip2 install pyhandle' returned a non-zero code: 1
```

**Comment by cookie33** [comment]
Was it fixed in #85 ?
